xgcm/grid.py

```diff
@@ -1049,20 +1060,32 @@ def interp(self, da, axis, metric_weighted=False, **kwargs):
         da_i : xarray.DataArray
             The interpolated data
         """
+        if isinstance(axis, list) or isinstance(axis, tuple):
+            # parse multi axis kwargs like `boundary`
+            multi_kwargs = {k: self._parse_axes_kwargs(v) for k, v in kwargs.items()}
+            out = da
+            for axx in axis:
+                out = self.interp(
+                    out, axx, **{k: v[axx] for k, v in multi_kwargs.items()}
+                )
+            return out
```

Comment:
This block should probably be factored out as well, since it is reused often

Reply:
I was thinking about that too, will do with the next step.

Reply:
I am not sure how to achieve this properly. Another class method, e.g. `mulit_axis_func(self, funcname)` which will grab e.g. `self.interp` and execute it over each axis? Or is this to convoluted?

Reply:
Thinking about it even the code below is very repetitive between e.g. `interp`/`max` etc (e.g. the metric weighted step). Is there a sleek way to factor this out? 

And should this be done now, or in a seperate PR?

Reply:
I am still not comfortable with how much repetition is introduced with this. We do need to refactor the code somehow. It needs to be done here, in this PR, because this PR is the one that introduced the pattern. Otherwise we build up a lot of technical debt.

I am not sure the best way to do it. A class method seems reasonable.

Reply:
I agree with you, and have started looking into it.